android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiWebChromeClient.java

```diff
@@ -172,5 +188,236 @@ public boolean interceptOnBackPressed()
 		}
 		return false;
 	}
+
+	// See: https://code.google.com/p/android/issues/detail?id=62220
+	// This is unsupported by Google Android
+	// openFileChooser for Android 3.0+
+	public void openFileChooser(ValueCallback<Uri> filePathCallback, String acceptType){
+
+	    if(mFilePathCallbackLegacy != null) {
+	        mFilePathCallbackLegacy.onReceiveValue(null);
+	    }
+	    mFilePathCallbackLegacy = filePathCallback;
+
+	    Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+	    if (takePictureIntent.resolveActivity(tiWebView.getProxy().getActivity().getPackageManager()) != null) {
+
+	        // Create the File where the photo should go
+	        File photoFile = null;
+	        try {
+	            photoFile = createImageFile();
+	            takePictureIntent.putExtra("PhotoPath", mCameraPhotoPath);
+	        } catch (IOException ex) {
+	            // Error occurred while creating the File
+	            Log.e(TAG, "Unable to create Image File", ex);
+	        }
+
+	        // Continue only if the File was successfully created
+	        if (photoFile != null) {
+	            mCameraPhotoPath = "file:" + photoFile.getAbsolutePath();
+	            mCameraPhotoUri = Uri.fromFile(photoFile);
+	            takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, mCameraPhotoUri);
+	        } else {
+	            takePictureIntent = null;
+	        }
+	    }
+
+	    Intent contentSelectionIntent = new Intent(Intent.ACTION_GET_CONTENT);
+	    contentSelectionIntent.addCategory(Intent.CATEGORY_OPENABLE);
+	    contentSelectionIntent.setType("image/*");
+
+	    Intent[] intentArray;
+	    if(takePictureIntent != null) {
+	        intentArray = new Intent[]{takePictureIntent};
+	    } else {
+	        intentArray = new Intent[0];
+	    }
+
+	    Intent chooserIntent = new Intent(Intent.ACTION_CHOOSER);
+	    chooserIntent.putExtra(Intent.EXTRA_INTENT, contentSelectionIntent);
+	    chooserIntent.putExtra(Intent.EXTRA_TITLE, "Image Chooser");
+	    chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, intentArray);
```

Comment:
this could be an empty array. I don't think we should be adding empty objects to the intent, unless it serves some purpose.